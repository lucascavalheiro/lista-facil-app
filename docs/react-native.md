# Getting Started with React Native

## Setting Up Your Machine for App Development

- iOS (You require a Mac to develop iOS Apps) Simply download XCode
- [Android on Mac](https://medium.com/pvtl/react-native-android-development-on-mac-ef7481f65e47#.w2hel88zy)
- Android on Windows - coming soon...

## Install React Native

[React Native - Get Started Guide](https://facebook.github.io/react-native/docs/getting-started.html) 

## Learn React

[React Native Express](http://www.reactnativeexpress.com/) is a great site to get you started, specifically:

- [Get your head around ES6](http://www.reactnativeexpress.com/es6)
- [What is JSX?](http://www.reactnativeexpress.com/jsx)
- [What are Components?](http://www.reactnativeexpress.com/components)
- [React State](http://www.reactnativeexpress.com/data_component_state)
- [Redux](http://www.reactnativeexpress.com/redux)

Once you've got your head around the basics, checkout the [React Native](https://facebook.github.io/react-native/) website, specifically:

- Go through ['The Basics'](https://facebook.github.io/react-native/docs/props.html)
- Gain an understanding of the [components](https://facebook.github.io/react-native/docs/activityindicator.html) React Native provides out of the box
