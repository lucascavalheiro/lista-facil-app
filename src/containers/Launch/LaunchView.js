/**
 * Launch Screen
 *  - Shows a nice loading screen whilst:
 *    - Preloading any specified app content
 *    - Checking if user is logged in, and redirects from there
 *
 * React Native Starter App
 * https://github.com/mcnamee/react-native-starter-app
 */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
  View,
  Image,
  Alert,
  StatusBar,
  StyleSheet,
  ActivityIndicator,
} from 'react-native';
import { Actions } from 'react-native-router-flux';

// Consts and Libs
import { AppStyles, AppSizes } from '@theme/';

/* Styles ==================================================================== */
const styles = StyleSheet.create({
  launchImage: {
    width: AppSizes.screen.width,
    height: AppSizes.screen.height,
  },
});

/* Component ==================================================================== */
class AppLaunch extends Component {
  static componentName = 'AppLaunch';

  static propTypes = {
    login: PropTypes.func.isRequired,
    getRecipes: PropTypes.func.isRequired,
    getMeals: PropTypes.func.isRequired,
  }

  componentDidMount = () => {
    // Show status bar on app launch
    StatusBar.setHidden(false, true);

    // Preload content here
    Promise.all([
      this.props.getMeals(),
      this.props.getRecipes(),
    ]).then(() => {
      // Once we've preloaded basic content,
      // - Try to authenticate based on existing token
      this.props.login()
        // Logged in, show index screen
        .then(() => Actions.app({ type: 'reset' }))
        // Not Logged in, show Login screen
        .catch(() => Actions.authenticate({ type: 'reset' }));
    }).catch(err => Alert.alert(err.message));
  }

  render = () => (
    <View style={[AppStyles.container]}>
      <Image
        source={require('../../images/launch.jpg')}
        style={[styles.launchImage, AppStyles.containerCentered]}
      >
        <ActivityIndicator
          animating
          size={'large'}
          color={'#C1C5C8'}
        />
      </Image>
    </View>
  );
}

/* Export Component ==================================================================== */
export default AppLaunch;
