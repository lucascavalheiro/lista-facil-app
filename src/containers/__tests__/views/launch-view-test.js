/**
 * Test to check if the component renders correctly
 */
/* global it expect */
import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import LaunchView from '@containers/Launch/LaunchView';

// Login prop expects a promise
const mockPromise = new Promise((resolve) => {
  resolve();
});

it('LaunchView renders correctly', () => {
  const tree = renderer.create(
    <LaunchView
      login={() => mockPromise}
      getRecipes={() => mockPromise}
      getMeals={() => mockPromise}
    />,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
