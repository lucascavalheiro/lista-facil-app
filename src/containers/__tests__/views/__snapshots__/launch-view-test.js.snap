// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LaunchView renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#E9EBEE",
        "flex": 1,
        "flexDirection": "column",
        "position": "relative",
      },
    ]
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "height": 1334,
          "width": 750,
        },
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
      ]
    }
  >
    <ActivityIndicator
      animating={true}
      color="#C1C5C8"
      hidesWhenStopped={true}
      size="large"
    />
  </Image>
</View>
`;
